#Хід роботи
1. Виконати демонстраційні приклади із даними, сформованими особисто.
2. Для одного набору даних (згідно варіанту) отримати оцінку середнього значення, моди, медіани, середньоквадратичного відхилення та дисперсії.
3. У висновках пояснити різницю між оцінками середнього значення, моди та медіани, дати смислову оцінку величині дисперсії.


In [91]:
import statistics
from fractions import Fraction as F
from decimal import Decimal as D
import random

In [114]:
# 1
print("Example 1")
data1 = [random.randint(1, 100) for x in range(9)]
mean_res = statistics.mean(data1)
frac_res = statistics.mean([F(3, 4), F(43, 56), F(5, 7), F(15, 2)])
dec_res = statistics.mean([D("3.4"), D("1.23"), D("66.99"), D("6.789")])
print(data1)
print(f"n\Mean:", mean_res, "\nFraction:", frac_res, "\nDecimal:", dec_res)

# 2
print("\nExample 2")
data_points = [random.randint(3, 50) for x in range(5, 745)]
dp_res1 = statistics.mean(data_points)
data_points2 = [random.triangular(7, 66, 90) for x in range(7, 3068)]
dp_res2 = statistics.mean(data_points2)
print("RandInt:", dp_res1, "\nTriangular:", dp_res2)

# 3
print("\nExample 3")
for i in range(3):
    data2 = [random.randint(1, 100) for x in range(1, 1001)]
    result = statistics.mode(data2)
    print(f"Mode result {i + 1}:", result)
animals = ["elephant", "lion", "lion", "elephant", "giraffe", "giraffe", "lion"]
result = statistics.mode(animals)
print("Mode result(animals):", result)

# 4
print("\nExample 4")
for i in range(3):
  data4 = [random.randint(6, 259) for x in range(768, 6969)]
  result4 = statistics.median(data4)
  print(f"Median {i + 1}", result4)
print("Median low:", statistics.median_low(data4))
print("Median high:",statistics.median_high(data4))

# 5
print("\nExample 5")
new_data = [random.randint(1, 99) for i in range(9)]
new_more_data = [random.randint(3, 7) for j in range(9)]
new_fractions = [F(7, 56), F(23, 4), F(7, 8)]

pvariance_result = statistics.pvariance(new_data)
pstdev_result = statistics.pstdev(new_data)
variance_result = statistics.variance(new_more_data)
stdev_result = statistics.stdev(new_more_data)
variance_fractions_result = statistics.variance(new_fractions)

print(f"PVariance Result: {pvariance_result}")
print(f"PStdev Result: {pstdev_result}")
print(f"Variance Result: {variance_result}")
print(f"Stdev Result: {stdev_result}")
print(f"Variance Fractions Result: {variance_fractions_result}")

Example 1
[13, 41, 60, 94, 95, 79, 34, 32, 90]
n\Mean: 59.77777777777778 
Fraction: 545/224 
Decimal: 19.60225

Example 2
RandInt: 26.728378378378377 
Triangular: 53.309070593323575

Example 3
Mode result 1: 94
Mode result 2: 37
Mode result 3: 25
Mode result(animals): lion

Example 4
Median 1 133
Median 2 131
Median 3 132
Median low: 132
Median high: 132

Example 5
PVariance Result: 1190.9876543209878
PStdev Result: 34.51068898647183
Variance Result: 3.361111111111111
Stdev Result: 1.8333333333333333
Variance Fractions Result: 597/64


In [99]:
# Input
course = [56, 58, 48, 50, 260, 5, 160, 125, 44, 16]
speed = [36, 250, 58, 100, 160, 163, 240, 158, 165, 50]
time = [2, 9, 24, 25, 28, 16, 18, 22, 6, 16]
# Calc
mean_course = statistics.mean(course)
mode_course = statistics.mode(course)
median_course = statistics.median(course)
stdev_course = statistics.stdev(course)
pvariance_course = statistics.pvariance(course)

mean_speed = statistics.mean(speed)
mode_speed = statistics.mode(speed)
median_speed = statistics.median(speed)
stdev_speed = statistics.stdev(speed)
pvariance_speed = statistics.pvariance(speed)

mean_time = statistics.mean(time)
mode_time = statistics.mode(time)
median_time = statistics.median(time)
stdev_time = statistics.stdev(time)
pvariance_time = statistics.pvariance(time)
# Output
print("Course:")
print("Mean:", mean_course)
print("Mode:", mode_course)
print("Median:", median_course)
print("Standard Deviation:", stdev_course)
print("PVariance:", pvariance_course)

print("\nSpeed:")
print("Mean:", mean_speed)
print("Mode:", mode_speed)
print("Median:", median_speed)
print("Standard Deviation:", stdev_speed)
print("PVariance:", pvariance_speed)

print("\nTime:")
print("Mean:", mean_time)
print("Mode:", mode_time)
print("Median:", median_time)
print("Standard Deviation:", stdev_time)
print("PVariance:", pvariance_time)

Course:
Mean: 82.2
Mode: 56
Median: 53.0
Standard Deviation: 78.01538309846335
PVariance: 5477.76

Speed:
Mean: 138
Mode: 36
Median: 159.0
Standard Deviation: 75.40851705506775
PVariance: 5117.8

Time:
Mean: 16.6
Mode: 16
Median: 17.0
Standard Deviation: 8.630694577430537
PVariance: 67.04


#ВИСНОВКИ
У ході виконання лабораторної роботи було проведено дослідження статистичних методів в мові програмування Python. Проведено аналіз трьох наборів даних, пов'язаних із курсом, швидкістю та часом руху літака. Застосовано модуль statistics для обчислення різних характеристик наборів даних.

Було виконано аналіз середнього значення, моди, медіани, середньоквадратичного відхилення та дисперсії для різних наборів даних. Для цього використовувались функції, такі як mean, mode, median, stdev, variance, pvariance, і pstdev.

Використання засобів Python, зокрема модулю statistics, робить обчислення статистичних показників зручним та ефективним.

У висновках важливо відзначити різницю між середнім значенням, модою та медіаною. Середнє значення є арифметичним середнім і враховує всі значення, в той час як мода представляє собою значення, що найчастіше зустрічається, а медіана - значення, розташоване по середині впорядкованого набору даних.

Смислова оцінка величини дисперсії важлива для розуміння ступеня розсіювання даних. У нашому випадку велика дисперсія вказує на значні відхилення від середнього значення, що може свідчити про різноманітність та нерівномірність розподілу. Величина дисперсії відображає ступінь коливань даних і може бути корисною для подальшого аналізу та прийняття рішень.

Course:

* Середнє значення (Mean): Середнє значення курсу складає 82.2 градуса.
* Мода (Mode): Найчастіше зустрічається значення курсу - 56 градусів.
* Медіана (Median): Медіанне значення курсу становить 53.0 градуса.
* Стандартне відхилення (Standard Deviation): Стандартне відхилення велике (78.02), що може вказувати на значні коливання значень навколо середнього.
* Дисперсія (Population Variance): Велика дисперсія (5477.76) підтверджує великі відхилення даних від середнього значення.

Speed:

* Середнє значення (Mean): Середня швидкість - 138 км/год.
* Мода (Mode): Найчастіше зустрічається значення швидкості - 36 км/год.
* Медіана (Median): Медіанна швидкість - 159.0 км/год.
* Стандартне відхилення (Standard Deviation): Стандартне відхилення становить 75.41, що також свідчить про значні коливання швидкостей.
* Дисперсія (Population Variance): Велика дисперсія (5117.8) підтверджує великі відхилення швидкостей від середнього значення.

Time:

* Середнє значення (Mean): Середній час руху - 16.6 хвилин.
* Мода (Mode): Значення моди - 16 хвилин.
* Медіана (Median): Медіанний час - 17.0 хвилин.
* Стандартне відхилення (Standard Deviation): Стандартне відхилення складає 8.63, що може вказувати на значні коливання часу руху.
* Дисперсія (Population Variance): Дисперсія (67.04) підтверджує великі відхилення часу руху від середнього значення.